### PR TITLE
Updated Setup Guide and Fixed Requirements

### PR DESCRIPTION
**Pull request by thisisayush**
- Added Missing Requirements (Pillow) to Requirements.txt
~- Created Missing Migrations~
- Added Setup Guide for development Environment in README.md

#1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+./venvconf/Scripts/activate
+```
+##### Linux
+```
+source venvconf/bin/activate
+```
+**Note:** To deactivate the Virtual Environment, type deactivate in the shell to deactivate.
+#### Clone your forked repo
+```
+git clone https://github.com/your_username/Confluence
```
Comment (CuriousLearner) [inline README.md]: Mention `<github_username>` instead of `your_username`

Also, add a note that people have to replace this with their github username

File: README.md [hunk 2]
```diff
@@ -1,2 +1,81 @@
...
+```
+##### Change to develop branch
+Code development should happed in 'develop' branch and not in 'master' branch
```
Comment (CuriousLearner) [inline README.md]: No, Development does not happen on develop branch. It would happen on a feature branch and PRs would be created from feature branch on their fork to the develop branch of this project.

Understand this, as, if many people are working on develop branch in their fork they have to continually rebase to get latest code and place their code on top of it (and solving merge conflicts).

We prefer solving merge conflicts when a single feature is complete (and also would be squashed to a single commit on develop branch)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,81 @@
...
+```
+git checkout develop
+```
+
+#### Installing Project Requirements (First Time Only or on Requirement changes)
+Change directory to cloned directory
```
Comment (CuriousLearner) [inline README.md]: Change to project dir, not cloned dir

File: README.md [hunk 4]
```diff
@@ -1,2 +1,81 @@
...
+```
+
+#### Installing Project Requirements (First Time Only or on Requirement changes)
+Change directory to cloned directory 
+```
+cd confluence
+```
+Use pip to install  missing requirements: (if you don't have pip installed on your system, google how to install) (Windows Users should run this command Adminsitrator Shell or will get error)
+```
+pip install -r  .\requirements.txt
```
Comment (CuriousLearner) [inline README.md]: no backward slashes here.

Simply goes by 

```pip install -r requirements.txt```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,81 @@
...
+```
+#### Creating Environment Variables File
+Change directory to src/confluence to create a ".env" file that will contain your enviroment variables. **Note:** This file will not ignored by git and will be available automatically on production environment.
```
Comment (CuriousLearner) [inline README.md]: `.env` does not get included in git. That is the whole point of keeping settings as environment variables (because we want to prevent including confidential information on git tree)

File: README.md [hunk 6]
```diff
@@ -1,2 +1,81 @@
...
+```
+cd src\confluence
+```
+Create file ".env" with following data:
+```
+EXPLARA_API_KEY=YOUR_EXPLARA_KEY_HERE
```
Comment (CuriousLearner) [inline README.md]: Change values to snake case with angular brackets like:

EXPLARA_API_KEY=<your_explara_key>

So that it looks different and someone reading the guide could quickly understand.

Also include a note to replace values with actual values.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,81 @@
...
+python manage.py makemigrations
+python manage.py migrate
+```
+
+#### Run Server 
+Once all above setup is done, run the following code to start your development server,
+```
+python manage.py runserver
+```
+You will see something like
```
Comment (CuriousLearner) [inline README.md]: `You will something like this` is not needed. 

Rest of the things are also not needed. 

Just add that you can visit the list at localhost:8000.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,81 @@
...
+```
+Performing system checks...
+
+System check identified no issues (0 silenced).
+June 16, 2017 - 23:46:33
+Django version 1.9, using settings 'confluence.settings'
+Starting development server at http://127.0.0.1:8000/
+Quit the server with CTRL-BREAK.
+```
+On successful setup you will see the second last line i.e. ' Starting Development Server at http://127.0.0.1:8000/ '. Visit this url on your web browser to see if server is started successfully. On failed startup, you will see a 'Refused to connect' error.
```
Comment (CuriousLearner) [inline README.md]: When would one get Refused to connect error?

**Review by CuriousLearner — CHANGES_REQUESTED** [review]
Hi @thisisayush !

Thanks for you work. I've mentioned my review inline.

Apart from that, please remove requirements.txt change, and all the migrations.

Migrations would be committed in the project at later point of time.

Apart from that, you're not supposed to commit your sqlite database, so please remove that.

Also, in the setup guide, add a create superuser command, with which developer can login to the admin panel.

File: README.md (hunk 8 above)
Comment (thisisayush) [reply README.md]: It happens when Server is not started i.e. the runserver command get stucked at some point while starting server due to some error and the developer may think that server is started.

**Comment by thisisayush** [comment]
Removing requirements.txt change will cause error while starting server because the social media app requires "Pillow" which wasn't available in requirements.txt and hence pip install -r requirements.txt will not install it and will require additional command to install in manually

**Comment by thisisayush** [comment]
Added Superuser Creation detail in README.md

File: README.md [hunk 9]
```diff
@@ -1,2 +1,88 @@
 # Confluence
 This is Confluence of Conference.
+
+# Setup Guide
+
+### Creating a Virtual Environment
+Virtual Environment allows us to have requirements installed specifically to our project instead of complete system. To create a virtual environment goto any directory where you will store your code and run the following command on terminal (linux) or Windows PowerShell (Windows)(For windows, run Windows Powershell as Administrator before running below commands)
+```
```
Comment (aktech) [inline README.md]: Please break lines after like 80-90 characters

Comment (aktech) [reply README.md]: goto -> go to

running below commands -> running following commands

File: README.md [hunk 10]
```diff
@@ -1,2 +1,88 @@
...
+git clone https://github.com/<github_username>/Confluence
+```
+##### Change to develop branch
+```
+git checkout develop
+```
+Development does not happen on develop branch. It would happen on a feature branch and PRs would be created from feature branch on their fork to the develop branch of this project.
+
+Understand this, as, if many people are working on develop branch in their fork they have to continually rebase to get latest code and place their code on top of it (and solving merge conflicts).
```
Comment (aktech) [inline README.md]: Please rephrase this to be more readable.

Comment (thisisayush) [reply README.md]: Umm?

**Comment by thisisayush** [comment]
Completed @aktech Requested Changes.
